test/unit/EcdsaPrivateKey.js

```diff
@@ -491,4 +493,120 @@ describe("EcdsaPrivateKey", function () {
+            // Set bodyBytes to null to simulate missing bodyBytes
+            mockedTransaction._signedTransactions = {
+                list: [{ bodyBytes: null }],
+            };
+
+            // Stub the _key.sign method to return a mock signature
+            privateKey._key = {
+                sign: sinon.stub().returns(mockedSignature),
+            };
+
+            // Call signTransaction method
+            const signatures = privateKey.signTransaction(mockedTransaction);
+
+            // Validate that an empty Uint8Array was returned
+            expect(signatures).to.deep.equal([new Uint8Array()]);
+
+            // Ensure that the transaction's addSignature method was called with the empty signature
+            sinon.assert.calledWith(
+                mockedTransaction.addSignature,
+                privateKey.publicKey,
+                [new Uint8Array()],
+            );
+
+            // Ensure that sign method of the privateKey._key was not called
+            sinon.assert.notCalled(privateKey._key.sign);
+        });
+
+        it("should sign transaction and add multiple signature", function () {
+            const mockedSignatures = [
+                new Uint8Array([10, 11, 12]),
+                new Uint8Array([13, 14, 15]),
+                new Uint8Array([16, 17, 18]),
+            ];
+
+            const signedTransactions = [
+                { bodyBytes: new Uint8Array([1, 2, 3]) },
+                { bodyBytes: new Uint8Array([4, 5, 6]) },
+                { bodyBytes: new Uint8Array([7, 8, 9]) },
+            ];
+
+            // Mock _signedTransactions.list to return an array of  transaction
+            mockedTransaction._signedTransactions = {
+                list: signedTransactions,
+            };
+
+            // Stub the _key.sign method to return a list of mock signature
+            privateKey._key = {
+                sign: sinon
+                    .stub()
+                    .onCall(0)
+                    .returns(mockedSignatures[0])
+                    .onCall(1)
+                    .returns(mockedSignatures[1])
+                    .onCall(2)
+                    .returns(mockedSignatures[2]),
+            };
+
+            // Call the real signTransaction method
+            const signatures = privateKey.signTransaction(mockedTransaction);
+
+            // Validate that the signatures are correct
+            expect(signatures).to.deep.equal(mockedSignatures);
+
+            // Ensure that the transaction's addSignature method was called with the correct arguments
+            sinon.assert.calledWith(
+                mockedTransaction.addSignature,
+                privateKey.publicKey,
+                mockedSignatures,
+            );
+
+            // Ensure that sign method of the privateKey._key was called the correct number of times
+            sinon.assert.callCount(privateKey._key.sign, 3);
+        });
+    });
```

Comment:
I wonder if we should create a new `PrivateKey` file with these unit tests in it. It just feels weird that we have this `PrivateKey` tests in the `EcdsaPrivateKey`.